### ISSUE TITLE
Null safety feedback: Warn on unnecessary import silence

### ISSUE THREAD

**Issue by kaladron** [dart-lang/sdk#45024]
In order to support the migration, packages can disable warnings:

// ignore: import_of_legacy_library_into_null_safe
import 'package:shared_preferences/shared_preferences.dart';

However, when packages migrate to null safety, these ignores are no longer needed.  Since Dart can detect when this is necessary, I'm hoping that it can detect when it's *not* necessary and warn on these (as is now the case with the latest update to shared_preferences).  This warning would be a good candidate for pedantic.

**Comment by bwilkerson** [comment]
We have support written for that, but have some internal clean-up that needs to happen before we can enable it.

The current plan is to enable the support always, not via lint rule, as we can't see any value for any user in having unnecessary ignore comments left in their code.

**Comment by srawlins** [comment]
Yeah duplicate of https://github.com/dart-lang/sdk/issues/35234 or https://github.com/dart-lang/sdk/issues/25917
